### ISSUE TITLE
同样的model和test_data，分布式环境auc打印，和本地计算auc值差别很大

### ISSUE THREAD

**Issue by jinyongbo128** [PaddlePaddle/Paddle#8138]

**Comment by jinyongbo128** [comment]
现象：
集群和本地，用同样的model和test_data，两者得到的auc差别很大，
集群可以达到0.9的auc，本地只有0.52

#集群auc打印方式：
def event_handler(event):
      if isinstance(event, paddle.event.EndIteration):
            if event.batch_id and event.batch_id % 10 == 0:
                 result = trainer.test(reader=test_reader, feeding=feeding)
                 print event.pass_id, event.batch_id, event.cost, result.cost, event.metrics

#本地auc打印方式：
1. 先从集群上getmerge下这个model，test_data
2. 使用paddle 的paddle.inference.Inference接口去打印predict_res和原来label信息
3. 使用auc脚本计算auc值

**Comment by QiJune** [comment]
集群上的训练模型的配置贴一下。
还有本地auc脚本贴一下呗，是不是本地计算错了

**Comment by jinyongbo128** [comment]
@QiJune

**Comment by jinyongbo128** [comment]
@QiJune   
learn_optimizer = paddle.optimizer.AdaGrad(
learning_rate=1e-2,
regularization=paddle.optimizer.L2Regularization(rate=1e-3*float(50)))

trainer = paddle.trainer.SGD(
cost=cost,
parameters=parameters,
extra_layers=paddle.evaluator.auc(input=output, label=lab),
update_equation=learn_optimizer,
is_local=False)

**Comment by QiJune** [comment]
本地的auc是怎么计算的，是不是本地计算错误？
